### DISCUSSION TITLE
llmware tests fail  even on fresh repo code

### DISCUSSION THREAD

**Discussion by virunew** [llmware-ai/llmware#237] [Q&A]
when I try to run ./run-tests.py -s to run all tests, I get 13 failures. The code is latest and llmware package version is 0.1.13. Error message is as below.  will appreciate any help on this  :

========================================================================================== short test summary info ==========================================================================================
FAILED embeddings/test_all_embedding_dbs.py::test_milvus_embedding_and_query - TypeError: EmbeddingHandler.delete_index() got an unexpected keyword argument 'model'
FAILED embeddings/test_all_embedding_dbs.py::test_faiss_embedding_and_query - TypeError: EmbeddingHandler.delete_index() got an unexpected keyword argument 'model'
FAILED embeddings/test_embeddings.py::test_milvus_embedding_and_query - TypeError: EmbeddingHandler.delete_index() got an unexpected keyword argument 'model'
FAILED embeddings/test_embeddings.py::test_faiss_embedding_and_query - TypeError: EmbeddingHandler.delete_index() got an unexpected keyword argument 'model'
FAILED embeddings/test_null_cases.py::test_no_blocks_faiss - llmware.exceptions.EmbeddingModelNotFoundException: Embedding model for 'test_no_blocks_faiss' could not be located
FAILED embeddings/test_null_cases.py::test_no_blocks_milvus - llmware.exceptions.EmbeddingModelNotFoundException: Embedding model for 'test_no_blocks_milvus' could not be located
FAILED embeddings/test_null_cases.py::test_no_blocks_pinecone - TypeError: expected string or bytes-like object, got 'NoneType'
FAILED embeddings/test_null_cases.py::test_no_blocks_mongo_atlas - pymongo.errors.OperationFailure: Unrecognized pipeline stage name: '$listSearchIndexes', full error: {'ok': 0.0, 'errmsg': "Unrecognized pipeline stage name: '$listSearchIndexes'", 'code': 40324, 'cod...
FAILED models/test_all_generative_models.py::test_all_gen_models - cohere.error.CohereError: No API key provided. Provide the API key in the client initialization or the CO_API_KEY environment variable.
FAILED models/test_cloud_model_providers.py::test_openai - AssertionError: assert 'artificial' in 'ai output not available'
FAILED models/test_cloud_model_providers.py::test_google - AssertionError: assert 'artificial' in 'ai output not available'
FAILED models/test_cloud_model_providers.py::test_anthropic - AssertionError: assert 'artificial' in 'ai output not available'
FAILED models/test_cloud_model_providers.py::test_ai21 - AssertionError: assert 'artificial' in 'ai output not available'
========================================================================== 13 failed, 16 passed, 14 warnings in 248.44s (0:04:08) ===========================================================================

**Comment by doberst** [comment]
Thanks for highlighting this - we are in the process of updating the automated test package.   A few of the tests rely upon endpoint dependencies (e.g., particular api_keys, etc) - so we are in the process of cleaning up and relaunching.   Will keep you updated on progress with this - and welcome any feedback and ideas along the way ...
